src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -52,26 +57,27 @@ const RichTextEditorComponent: FC<RichTextEditorProps> = ({
     }, [clear]);
 
     const onChange = useCallback(
-        debounce((value: TNode[]) => setDebouncedValue(value), ON_SAVE_DELAY_IN_MS),
+        debounce((value: TNode[]) => {
+            setDebouncedValue(value);
+        }, ON_SAVE_DELAY_IN_MS),
         [],
     );
 
     return (
         <div data-test-id="rich-text-editor" className="tw-relative tw-w-full">
             <Plate
-                initialValue={parseRawValue(initialValue)}
-                onChange={onChange}
+                id={editorId}
+                editor={editor}
+                initialValue={parseRawValue(editor, initialValue)}
+                onChange={(value) => {
+                    onChange(value);
+                    localValue.current = value;
```

Comment:
might be worth moving this line into the useCallback so that the function is not re-rendered each time